lua/neogen/init.lua

```diff
@@ -149,128 +125,12 @@ neogen.configuration = {
-            located_parent_node,
-            data,
-            language.template,
-            opts.type
-        )
-
-        if #content ~= 0 then
-            cursor.del_extmarks() -- Delete previous extmarks before setting any new ones
-
-            local jump_text = language.jump_text or neogen.configuration.jump_text
-
-            --- Removes jump_text marks and keep the second part of jump_text|other_text if there is one (which is other_text)
-            local delete_marks = function(v)
-                local pattern = jump_text .. "[|%w]+"
-                local matched = string.match(v, pattern)
-
-                if matched then
-                    local split = vim.split(matched, "|", true)
-                    if #split == 2 and neogen.configuration.input_after_comment == false then
-                        return string.gsub(v, jump_text .. "|", "")
-                    end
-                else
-                    return string.gsub(v, jump_text, "")
-                end
-
-                return string.gsub(v, pattern, "")
-            end
-
-            local content_with_marks = vim.deepcopy(content)
-
-            -- delete all jump_text marks
-            content = vim.tbl_map(delete_marks, content)
-
-            -- Append the annotation in required place
-            vim.fn.append(to_place, content)
-
-            -- Place cursor after annotations and start editing
-            -- First and last extmarks are needed to know the range of inserted content
-            if neogen.configuration.input_after_comment == true then
-                -- Creates extmark for the beggining of the content
-                cursor.create(to_place + 1, start_column)
-                -- Creates extmarks for the content
-                for i, value in pairs(content_with_marks) do
-                    local start = 0
-                    local count = 0
-                    while true do
-                        start = string.find(value, jump_text, start + 1)
-                        if not start then
-                            break
-                        end
-                        cursor.create(to_place + i, start - count * #jump_text)
-                        count = count + 1
-                    end
-                end
-
-                -- Create extmark to jump back to current location
-                local pos = vim.api.nvim_win_get_cursor(0)
-                local col = pos[2] + 2
-
-                -- If the line we are in is empty, it will throw an error out of bounds.
-                if vim.api.nvim_get_current_line() == "" then
-                    col = pos[2] + 1
-                end
-
-                cursor.create(pos[1], col)
-
-                -- Creates extmark for the end of the content
-                cursor.create(to_place + #content + 1, 0)
-
-                cursor.jump({ first_time = true })
-            end
-        end
-    end
+    require("neogen.generator")(vim.bo.filetype, opts and opts.type)
```

Comment:
@kevinhwang91 Lua lsp is throwing me a warning about the function that accepts 0 variables. Maybe it's because of the `__call`. Is it possible to stop the warning maybe by making the return of the require a function that accepts 2 variables, or something else ?